plugins/field-colour-hsv-sliders/src/field_colour_hsv_sliders.js

```diff
@@ -0,0 +1,639 @@
+        Blockly.browserEvents.conditionalBind(
+            this.hueSlider_, 'input', this, this.onSliderChange_));
+    this.boundEvents_.push(
+        Blockly.browserEvents.conditionalBind(
+            this.saturationSlider_, 'input', this, this.onSliderChange_));
+    this.boundEvents_.push(
+        Blockly.browserEvents.conditionalBind(
+            this.brightnessSlider_, 'input', this, this.onSliderChange_));
+
+    container.appendChild(document.createElement('hr'));
+
+    if (window.EyeDropper) {
+      // If the browser supports the eyedropper API, create a button for it.
+      const button = document.createElement('button');
+      button.classList.add('fieldColourEyedropper');
+      container.appendChild(button);
+      this.boundEvents_.push(
+          Blockly.browserEvents.conditionalBind(
+              button, 'click', this, this.onEyedropperEvent_));
+    } else {
+      // Otherwise, fall back on colour input elements. Browsers generally
+      // provide an eyedropper feature as part of their colour input editor.
+      const label = document.createElement('label');
+      label.classList.add('fieldColourEyedropper');
+      const input = document.createElement('input');
+      input.type = 'color';
+      label.appendChild(input);
+      container.appendChild(label);
+      this.boundEvents_.push(
+          Blockly.browserEvents.conditionalBind(
+              input, 'input', this, this.onBrowserColourInputEvent_));
+    }
+
+    this.dropdownContainer_ = container;
+
+    this.updateSliderValues_();
+  }
+
+  /**
+   * Disposes of events and DOM-references belonging to the colour editor.
+   * @private
+   */
+  dropdownDisposeSliders_() {
+    for (const event of this.boundEvents_) {
+      Blockly.browserEvents.unbind(event);
+    }
+    this.boundEvents_.length = 0;
+    this.hueReadout_ = null;
+    this.hueSlider_ = null;
+    this.saturationReadout_ = null;
+    this.saturationSlider_ = null;
+    this.brightnessReadout_ = null;
+    this.brightnessSlider_ = null;
+    this.dropdownContainer_ = null;
+  }
+
+  /**
+   * Updates the value of this field based on the editor sliders.
+   * @param {?Event} event Unused.
+   * @private
+   */
+  onSliderChange_(event) {
+    const hue = parseFloat(this.hueSlider_.value);
+    const saturation = parseFloat(this.saturationSlider_.value);
+    const brightness = parseFloat(this.brightnessSlider_.value);
+    const hsv = new HsvColour(
+        hue / 360.0, saturation / 100.0, brightness / 100.0);
+    const rgb = new RgbColour().loadFromHsv(hsv);
+    this.setValue(rgb.toHex());
+    this.renderSliders_();
+  }
+
+  /**
+   * Updates the value of this field and editor sliders using an eyedropper.
+   * @param {?Event} event Unused.
+   * @private
+   */
+  onEyedropperEvent_(event) {
+    const eyeDropper = new window.EyeDropper();
+    eyeDropper.open().then((result) => {
+      this.setValue(result.sRGBHex);
+      this.updateSliderValues_();
```

Comment:
Instead of calling this explicitly, I think you can use `doValueUpdate_`.

Reply:
Done.